src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -1528,6 +1528,10 @@ void handle_actor_worker_disconnect(LocalSchedulerState *state,
 
   /* Attempt to dispatch some tasks because some resources may have freed up. */
   dispatch_all_tasks(state, algorithm_state);
+
+  /* Start a worker to replace the removed actor's worker and replenish the
+   * worker pool. */
+  start_worker(state);
```

Comment:
This is probably a good idea, but note that it does prevent the node from "scaling down". That is, once we create 1000 actors on a node, there is no going back.

It may be worth making this change in xray as well (in `node_manager.cc`). This code will be deprecated soon, so that could save us the trouble of tracking this down a second time.

Reply:
A solution could be to store the initial number of workers in the local scheduler state and only replenish if available_workers + active_actors < initial_workers.

Alternatively, there could be a distinction between actor workers and task workers so that the number of available worker is independent from the number of actors active. In other words, starting a new actor will launch a new worker as well to replenish the taken worker.

Reply:
if we go with this solution, wouldn't it be better to call `start_worker` before `dispatch_all_tasks`? If the dispatch queue is not empty and the number of available workers is zero, `dispatch_tasks` will start a worker. Then another worker will be started with the `start_worker(state)` line added above.

Reply:
Yes, good catch. Should we revert the change and I'll make the update, or should I open a new PR to fix the bug?

I hope we can find a more elegant solution for a future release to solve the scaling down issue.